data-collection/src/main/java/refactoringml/RefactoringAnalyzer.java

```diff
@@ -33,49 +29,52 @@
 	private Repository repository;
 	private boolean storeFullSourceCode;
 	private String fileStorageDir;
+	private PMDatabase pmDatabase;
 
 	private static final Logger log = LogManager.getLogger(RefactoringAnalyzer.class);
 
-	public RefactoringAnalyzer (Project project, Database db, Repository repository, String fileStorageDir, boolean storeFullSourceCode) {
+	public RefactoringAnalyzer (Project project, Database db, Repository repository, PMDatabase pmDatabase, String fileStorageDir, boolean storeFullSourceCode) {
 		this.project = project;
 		this.db = db;
 		this.repository = repository;
 		this.storeFullSourceCode = storeFullSourceCode;
-
 		this.tempDir = null;
+		this.pmDatabase = pmDatabase;
 		this.fileStorageDir = lastSlashDir(fileStorageDir);
 	}
 
-	public List<RefactoringCommit> collectCommitData(RevCommit commit, CommitMetaData superCommitMetaData, List<Refactoring> refactoringsToProcess) throws IOException {
+	public List<RefactoringCommit> collectCommitData(RevCommit commit, CommitMetaData superCommitMetaData, List<Refactoring> refactoringsToProcess, List<DiffEntry> entries) {
 		List<RefactoringCommit> allRefactorings = new ArrayList<>();
 
 		try {
-			// get the map between new path -> old path
-			HashMap<String, String> filesMap = getMapWithOldAndNewFiles(repository, commit);
+			//TODO: is this obsolete? JGit misses many files, e.g. after move source dir refactorings
```

Comment:
This is required because sometimes RefactoringMiner returns only the "new" file names. In such cases, we need the map of the old file names in order to match things (and that's why we use the git info).

Reply:
Thanks for the explanation.